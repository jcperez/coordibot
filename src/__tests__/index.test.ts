import * as moment from "moment";
import * as Coordibot from "../index";

jest.mock("../adapters/gcalendar");

describe("Handlers", () => {
  beforeAll(() => {
    console.error = () => {};
    console.log = () => {};
  });

  test("Retrieve availability", async () => {
    const availabilityTest = (date: string) => {
      return new Promise((resolve) => {
        Coordibot.retrieveAvailabilityHandler({
          messageVersion: "1.0",
          invocationSource: "FulfillmentCodeHook",
          userId: "user-1",
          sessionAttributes: {},
          bot: {
            name: "CoordinatorBot",
            alias: "$LATEST",
            version: "$LATEST",
          },
          outputDialogMode: "Text",
          currentIntent: {
            name: "ScheduleInterview",
            slots: {
              interviewDate: date,
            },
            confirmationStatus: "None",
          },
        }, null, (err: any, param: any) => {
          resolve(param);
        });
      });
    };

    const today = moment(1527200000000).format("YYYY-MM-DD");
    const response: any = await availabilityTest(today);
    return expect(response).toMatchSnapshot();
  });

  test("Connection checker", async () => {
    const connectionTest = () => {
      return new Promise((resolve) => {
        Coordibot.testRetrieveAvailabilityHandler({
          messageVersion: "1.0",
          invocationSource: "FulfillmentCodeHook",
          userId: "user-1",
          sessionAttributes: {},
          bot: {
            name: "CoordinatorBot",
            alias: "$LATEST",
            version: "$LATEST",
          },
          outputDialogMode: "Text",
          currentIntent: {
            name: "ScheduleInterview",
            confirmationStatus: "None",
          },
        }, null, (err: any, param: any) => {
          resolve(param);
        });
      });
    };

    const response: any = await connectionTest();
    return expect(response).toMatchSnapshot();
  });
});
