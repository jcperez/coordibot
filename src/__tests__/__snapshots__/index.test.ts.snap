// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Handlers Connection checker 1`] = `
Object {
  "dialogAction": Object {
    "fulfillmentState": "Fulfilled",
    "message": Object {
      "content": "juan.carlos@wizeline.com - OK
",
      "contentType": "PlainText",
    },
    "type": "Close",
  },
  "sessionAttributes": Object {},
}
`;

exports[`Handlers Retrieve availability 1`] = `
Object {
  "dialogAction": Object {
    "fulfillmentState": "Fulfilled",
    "message": Object {
      "content": "Availability for 2018-05-25:
*10:00 to 12:00* - juan.carlos@wizeline.com - Hold for interview

*14:00 to 15:00* - juan.carlos@wizeline.com - Hold for interview
 - Event duration is less than 90 mins

",
      "contentType": "PlainText",
    },
    "type": "Close",
  },
  "sessionAttributes": Object {},
}
`;
